### PR TITLE
added get_output_shape function

### PR DESCRIPTION
**Pull request by priyatharsan**
@rfratila

### CODE REVIEW

File: vulcanai2/tests/models/test_layers.py [hunk 1]
```diff
@@ -3,25 +3,17 @@
 import torch
 from vulcanai2.models.layers import *
 
-class TestBaseLayer:
+class TestBaseUnit:
     @pytest.fixture
-    def baselayer(self): 
-        return BaseLayer()
-
-    # TODO: Needs fixing assigning values to NoneType
```
Comment (rfratila) [inline vulcanai2/tests/models/test_layers.py]: Why was this deleted?

File: vulcanai2/models/utils.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+def get_size(summary_dict, output):
+    if isinstance(output, tuple):
+        for i in xrange(len(output)):
+            summary_dict[i] = OrderedDict()
+            summary_dict[i] = get_size(summary_dict[i],output[i])
+    else:
+        summary_dict['output_shape'] = list(output.size())
+    return summary_dict
+ 
+def get_output_shapes(input_size, model):
```
Comment (rfratila) [inline vulcanai2/models/utils.py]: I would like to see more documentation here since it seems quite complicated to extract the output shapes of each layer. Also if a function is in a class, we shouldn't need to pass it a model since it should just look at `self`. it doesn't make sense to use it as a static class when we've already instantiated it in the script that it is made it.

File: vulcanai2/tests/models/test_layers.py (hunk 1 above)
Comment (priyatharsan) [reply vulcanai2/tests/models/test_layers.py]: Rename to `BaseUnit`

Comment (rfratila) [reply vulcanai2/tests/models/test_layers.py]: sorry, I was referring to the #TODO comment below

Comment (rfratila) [reply vulcanai2/tests/models/test_layers.py]: the test function of modifying variables in baseunit

File: vulcanai2/models/utils.py (hunk 2 above)
Comment (priyatharsan) [reply vulcanai2/models/utils.py]: Tbh it was quite complicate for me to understand as well. 
ref links: https://pytorch.org/tutorials/beginner/former_torchies/nn_tutorial.html#forward-and-backward-function-hooks
https://github.com/sksq96/pytorch-summary/blob/master/torchsummary/torchsummary.py

File: vulcanai2/tests/models/test_layers.py (hunk 1 above)
Comment (priyatharsan) [reply vulcanai2/tests/models/test_layers.py]: those variables are dummy and are not being used. Although I would like to take a look at them once while working on the other layer classes.

File: vulcanai2/models/utils.py (hunk 2 above)
Comment (rfratila) [reply vulcanai2/models/utils.py]: ok for now let's put a link to this resource in the code and modify the function signature to just use the internal model instead of having to pass it one

File: vulcanai2/tests/models/test_layers.py (hunk 1 above)
Comment (rfratila) [reply vulcanai2/tests/models/test_layers.py]: ok let's make a not of that through docs in the BaseUnit class

File: vulcanai2/models/utils.py (hunk 2 above)
Comment (priyatharsan) [reply vulcanai2/models/utils.py]: so in that case it has to be moved to the `BaseNetwork`, right?

Comment (rfratila) [reply vulcanai2/models/utils.py]: yeah good idea. it will be used in all networks

File: vulcanai2/models/__init__.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 from .basenetwork import BaseNetwork
 from .cnn import ConvNet
 from .dnn import DenseNet
+from .utils import *
```
Comment (nirtiac) [inline vulcanai2/models/__init__.py]: Why do we have this? I'm worried about "polluting the namespace", especially if our utility functions are generically named.

Comment (rfratila) [reply vulcanai2/models/__init__.py]: that is a good point. @priyatharsan would you be able to specify what from utils you require so we can be more specific with the imports?

Comment (rfratila) [reply vulcanai2/models/__init__.py]: also on that note, I think what is found in utils should be tied to BaseNetwork since intuitively you should be able to do model.get_size and such instead of passing a model to get_size(model) and get_output_shapes(model)? I'm curious to know your thoughts on this matter

Comment (nirtiac) [reply vulcanai2/models/__init__.py]: Oh I didn't notice that. Good point, I agree completely, those should be with BaseNetwork.

Comment (priyatharsan) [reply vulcanai2/models/__init__.py]: removed the contents of utils to BaseNetwork as it makes more sense there

Comment (rfratila) [reply vulcanai2/models/__init__.py]: feel free to push your commit to this branch so we can look at the updated version

**Comment by priyatharsan** [comment]
I'm merging the branch for now. Getting few conflicts locally.

**Comment by rfratila** [comment]
can we delete the branch?
